### ISSUE TITLE
A slight incompatibility of Gradle 4.0 with IDEA

### ISSUE THREAD

**Issue by blindpirate** [gradle/gradle#2315]
Gradle 4.0 will produce classes into separated directory for each JVM language, which is slightly incompatible with IDEA. 

### Expected Behavior

Compatible with IDEA

### Current Behavior

Slightly incompatible with IDEA

### Context & Steps to Reproduce (for bugs)

Suppose I have a simple project:

```
src/main/java/a/A.java
src/test/groovy/a/ATest.groovy
```

and `java`/`groovy` plugins applied.

I run `gradle test` with Gradle 4.0, all works well. But when I import this project into IDEA, I notice `test module`'s test output path is `build/classes/java/test`:

![image](https://user-images.githubusercontent.com/12689835/27216305-c063dc58-52a7-11e7-9143-0235fc107cb6.png)

![image](https://user-images.githubusercontent.com/12689835/27216362-09772814-52a8-11e7-9da0-219a3201b1ea.png)

The consequence is, at this point, if I develop in IDEA and right click `ATest.groovy` -> `Run ATest`, IDEA will compile and generate the class into `build/classes/java/test`:

![image](https://user-images.githubusercontent.com/12689835/27216417-432f7a02-52a8-11e7-93eb-0c6e3f726fdd.png)

Now, I run `gradle test` and I get two identical tests:

![image](https://user-images.githubusercontent.com/12689835/27216441-5e0b9efa-52a8-11e7-8460-e7f9d7dfa4bc.png)

I tried to find a simple solution but failed. It seems that IDEA only supports one test output path for each module, no matter how many language directories there are.

A workaround is always run `clean` before `test`.

### Your Environment

It's easy to reproduce, and has nothing to do with environment.

**Comment by oehme** [comment]
@blindpirate does this also happen if you generate the project using `gradle idea` or only if you use the IDEA importer?

**Comment by blindpirate** [comment]
Well, even when I use `gradle idea` to generate the project, usually I need to click `import Gradle projects` or `refresh` in IDEA and then this occurs.

**Comment by oehme** [comment]
No the two are mutually exclusive. If you use `gradle idea`  you should use "Open Project". If you use "Import Gradle Project". Then you don't need `gradle idea`.

Can you please try the former and let me know if the output paths are okay? Next, can you try Ideas importer with and without the "Module for each SourceSet" option?

**Comment by oehme** [comment]
I tried it myself and everything works fine when running `gradle idea`. A completely separate output directory is configured. The IDEA importer uses a different default, which is broken with 4.0.

Until that is fixed in IDEA, you can work around it with this setting:

```
idea.module.inheritOutputDirs = true
```

**Comment by ijuma** [comment]
@oehme, thanks that's good to know. Has an issue been filed in IDEA's bug tracker?

**Comment by blindpirate** [comment]
@oehme Sorry for the late response, I'm on a vocation in the past few days. Happy to know you've already figure out this. So do you mean there's nothing Gradle can do? if so, I'll close this.

**Comment by oehme** [comment]
I already talked to @vladsoroka and this will be fixed in IntelliJ. There's indeed nothing to do on the Gradle side, except pointing out the workaround above.

**Comment by childnode** [comment]
@oehme do you know in which intelliJ release the patch will be included? Is there a superseding youTrack issue for that?

**Comment by oehme** [comment]
@vladsoroka will be able to answer that

**Comment by vladsoroka** [comment]
@childnode @oehme the change should be available since IDEA 2017.2 EAP 172.3198.6  (27th of June, 2017)
